### PR TITLE
(1) templatize apache configs (2) add MIME expire defaults

### PR DESCRIPTION
**Pull request by cdaringe**

### CODE REVIEW

File: roles/phpserver/templates/coins-ssl_vhost_default.j2 [hunk 1]
```diff
@@ -1,98 +1,53 @@
+{% extends "coins-apache-base.j2" %}
+{%- block vhost_entry %}
+
 <IfModule mod_ssl.c>
```
Comment (cdaringe) [inline roles/phpserver/templates/coins-ssl_vhost_default.j2]: should we really even be checking this? or should we just be flipping on the SSL engine and erring out if not present?

Comment (dylancwood) [inline roles/phpserver/templates/coins-ssl_vhost_default.j2]: Good point. I think it is just a line from the default config...

File: ansible.cfg [hunk 2]
```diff
@@ -30,7 +30,7 @@ module_lang    = C
 # smart - gather by default, but don't regather if already gathered
 # implicit - gather by default, turn off with gather_facts: False
 # explicit - do not gather by default, must say gather_facts: True
-gathering = implicit
+gathering = smart
```
Comment (dylancwood) [inline ansible.cfg]: won't you miss saying `gathering facts` all the time?

File: roles/phpserver/templates/coins-apache-base.j2 [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        RewriteRule ^/?portals/(.+)Portal/scripts/?(.*)$ /portals/master_portal/scripts/$2
+        RewriteRule ^/?portals/(.+)Portal/graphics/?(.*)$ /portals/master_portal/graphics/$2
+    </Directory>
+
+    {%- endblock document_root %}
+
+
+    {%- block api %}
+
+    <Directory "/var/www/html/coins_core/api">
```
Comment (dylancwood) [inline roles/phpserver/templates/coins-apache-base.j2]: If we are going to use this for COINS3.0 servers, then I think that all the paths should be `/coins/www/html`

**Comment by dylancwood** [comment]
You're using templates like a boss! Just fix the `/var/ww` vs `/coins/www` and we'll be good to go

File: ansible.cfg (hunk 2 above)
Comment (cdaringe) [inline ansible.cfg]: DONT WORRY, it still says it most of the time :)

**Pull request merged**